src/Illuminate/Mail/Mailer.php

```diff
@@ -553,41 +546,19 @@ protected function dispatchSentEvent($message, $data = [])
     {
         if ($this->events) {
             $this->events->dispatch(
-                new MessageSent($message->getSwiftMessage(), $data)
+                new MessageSent($message->getSymfonyMessage(), $data)
             );
         }
     }
 
     /**
-     * Force the transport to re-connect.
-     *
-     * This will prevent errors in daemon queue situations.
-     *
-     * @return void
-     */
-    protected function forceReconnection()
-    {
-        $this->getSwiftMailer()->getTransport()->stop();
-    }
```

Comment:
`stop()' in Symfony Mailer which is currently private
https://github.com/symfony/mailer/blob/c1f83da2296741110be35dd779f2a9e412cec466/Transport/Smtp/SmtpTransport.php#L258

This is called only from three places:
1. Destructing the Object
https://github.com/symfony/mailer/blob/c1f83da2296741110be35dd779f2a9e412cec466/Transport/Smtp/SmtpTransport.php#L356
2. Inside the `checkRestartThreshold()` method (Stop connection after a specific amount of messages was send and then sleep for a given amount of time.)
https://github.com/symfony/mailer/blob/c1f83da2296741110be35dd779f2a9e412cec466/Transport/Smtp/SmtpTransport.php#L333
3. Or if the `ping()` method is called (to detect timeouts) and in there throws an TransportException (which means we got a timeout)
https://github.com/symfony/mailer/blob/c1f83da2296741110be35dd779f2a9e412cec466/Transport/Smtp/SmtpTransport.php#L285

The only way I see it is possible to force a Reconnection is with a PR to Symfony Mailer to make the method public. Also this is SMTP Specific code and is not available on an interface.

Or another way may be to construct a new Transport Object. (Probably not intended)

Is there a specific use case for this forcedReconnection?

Reply:
Thanks @Jubeki. The only note I could find was the one from above in the docblock:

> This will prevent errors in daemon queue situations.

Reply:
So I guess for long running processes where the mail connection has been established already.

Reply:
This was introduced in Laravel 5.2 with the following PR: https://github.com/laravel/framework/pull/13583
It seems long working processes can throw SSL-Errors when trying to send. I didn't quite get the reason why though.

I don't know if this Problem will occur again but I think it will not because:
Before sending a message there is a check if 100s (configurable) have passed since the last message was sent to the SMTP Server. If 100s have passed it will first ping the SMTP-Server with a `NOOP` command to check the Connection. If an error occurs it will be reset and will only send the message then.
If after the reset an error occurs while sending the message, it should be a permanent error.

Reply:
Thanks @Jubeki

Reply:
Can't we force it by destroying the object and creating a new one?

Reply:
Yes that is possible. Could something unexpected happen if something relies on the same instance?
But who otherwise would call `forceReconnection` right?

Reply:
how can i force reconnection in laravel9? i having the problem with Unable to write bytes on the wire and SSL: Connection reset by peer if its run on supervisor daemon